test/unit/iam-assume-authenticator.test.js

```diff
@@ -0,0 +1,138 @@
+  const config = {
+    apikey: 'some-key',
+    iamProfileId: 'some-id',
+    url: 'iam.staging.com',
+    clientId: 'my-id',
+    clientSecret: 'my-secret',
+    disableSslVerification: true,
+    headers: {
+      'X-My-Header': 'some-value',
+    },
+    scope: 'A B C D',
+  };
+
+  it('should store all config options on the class', () => {
+    const authenticator = new IamAssumeAuthenticator(config);
+
+    expect(authenticator.authenticationType()).toEqual(Authenticator.AUTHTYPE_IAM_ASSUME);
+    expect(authenticator.url).toBe(config.url);
+    expect(authenticator.clientId).toBe(config.clientId);
+    expect(authenticator.clientSecret).toBe(config.clientSecret);
+    expect(authenticator.disableSslVerification).toBe(config.disableSslVerification);
+    expect(authenticator.headers).toEqual(config.headers);
+    expect(authenticator.scope).toEqual(config.scope);
+
+    // Should also create a token manager. Note that the options like `iamProfileId`
+    // aren't stored in the authenticator, but in the token manager.
+    expect(authenticator.tokenManager).toBeInstanceOf(IamAssumeTokenManager);
+    expect(authenticator.tokenManager.iamProfileId).toBe(config.iamProfileId);
+  });
+
+  it('should throw an error when no api key is provided', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator();
+    }).toThrow('Missing required parameters: apikey');
+  });
+
+  it('should throw an error when no profile information is provided', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({ apikey: config.apikey });
+    }).toThrow(
+      'Exactly one of `iamProfileName`, `iamProfileCrn`, or `iamProfileId` must be specified.'
+    );
+  });
+
+  it('should throw an error when too much profile information is provided', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({
+        apikey: config.apikey,
+        iamProfileId: 'some-id',
+        iamProfileCrn: 'some-crn',
+      });
+    }).toThrow(
+      'Exactly one of `iamProfileName`, `iamProfileCrn`, or `iamProfileId` must be specified.'
+    );
+  });
+
+  it('should throw an error when a profile name is provided without an account id', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({
+        apikey: config.apikey,
+        iamProfileName: 'some-id',
+      });
+    }).toThrow('`iamProfileName` and `iamAccountId` must be provided together, or not at all');
+  });
+
+  it('should throw an error when an account id is provided without a profile name', () => {
+    expect(() => {
+      const unused = new IamAssumeAuthenticator({
+        apikey: config.apikey,
+        iamProfileId: 'some-id',
+        iamAccountId: 'some-account',
+      });
+    }).toThrow('`iamProfileName` and `iamAccountId` must be provided together, or not at all');
+  });
+
+  // "End to end" style test, to make sure this authenticator ingregates properly with parent classes.
```

Comment:
```suggestion
  // "End to end" style test, to make sure this authenticator integrates properly with parent classes.
```
Sorry, but with @pyrooka on the review list, I need to demonstrate that I'm paying attention too :joy:

Reply:
Hard to see what he misses 😂 good catch!

Reply:
> Sorry, but with @pyrooka on the review list, I need to demonstrate that I'm paying attention too 😂

I left some for you on purpose, hahaha!